test/coap_option_test.dart

```diff
@@ -87,86 +86,21 @@ void main() {
+      final opt1 = ContentFormatOption(oneByteValue);
+
+      final opt2 = ContentFormatOption(twoByteValue);
+
+      final opt3 = ContentFormatOption(twoByteValue);
 
       expect(opt1 == opt2, isFalse);
       expect(opt2 == opt3, isTrue);
     });
 
-    test('Set string value', () {
-      final option = CoapOption.create(OptionType.fromTypeNumber(11))
-        ..stringValue = '';
-      expect(option.length, 0);
-
-      option.stringValue = 'CoAP.NET';
-      expect(option.stringValue, 'CoAP.NET');
-    });
-
-    test('Set int value', () {
-      final option = CoapOption.create(OptionType.fromTypeNumber(12))
-        ..intValue = 0;
-      expect(option.byteValue[0], 0);
-
-      option.intValue = 11;
-      expect(option.byteValue[0], 11);
-
-      option.intValue = 255;
-      expect(option.byteValue[0], 255);
-
-      option.intValue = 256;
-      expect(option.byteValue[0], 0);
-      expect(option.byteValue[1], 1);
-
-      option.intValue = 18273;
-      expect(option.byteValue[0], 97);
-      expect(option.byteValue[1], 71);
-
-      option.intValue = 1 << 16;
-      expect(option.byteValue[0], 0);
-      expect(option.byteValue[1], 0);
-      expect(option.byteValue[2], 1);
-
-      option.intValue = 23984773;
-      expect(option.byteValue[0], 133);
-      expect(option.byteValue[1], 250);
-      expect(option.byteValue[2], 109);
-      expect(option.byteValue[3], 1);
-
-      option.intValue = 0xFFFFFFFF;
-      expect(option.byteValue[0], 0xFF);
-      expect(option.byteValue[1], 0xFF);
-      expect(option.byteValue[2], 0xFF);
-      expect(option.byteValue[3], 0xFF);
-
-      // ignore: avoid_js_rounded_ints
-      option.intValue = 0x9823749837239845;
-      expect(option.byteValue[0], 69);
-      expect(option.byteValue[1], 152);
-      expect(option.byteValue[2], 35);
-      expect(option.byteValue[3], 55);
-      expect(option.byteValue[4], 152);
-      expect(option.byteValue[5], 116);
-      expect(option.byteValue[6], 35);
-      expect(option.byteValue[7], 152);
-
-      option.intValue = 0xFFFFFFFFFFFFFFFF;
-      expect(option.byteValue[0], 0xFF);
-      expect(option.byteValue[1], 0xFF);
-      expect(option.byteValue[2], 0xFF);
-      expect(option.byteValue[3], 0xFF);
-      expect(option.byteValue[4], 0xFF);
-      expect(option.byteValue[5], 0xFF);
-      expect(option.byteValue[6], 0xFF);
-      expect(option.byteValue[7], 0xFF);
-    });
-
     test('Join', () {
-      final opt1 = CoapOption.createString(OptionType.uriPath, 'Hello');
-      final opt2 = CoapOption.createString(OptionType.uriPath, 'from');
-      final opt3 = CoapOption.createString(OptionType.uriPath, 'me');
-      final str = CoapOption.join(<CoapOption>[opt1, opt2, opt3], '/');
+      final opt1 = UriPathOption('Hello');
+      final opt2 = UriPathOption('from');
+      final opt3 = UriPathOption('me');
+      final str = Option.join(<Option<Object?>>[opt1, opt2, opt3], '/');
```

Comment:
Tiny nitpick: I think we could omit the type for the list here and in other places, since it can be inferred?

```dart
final str = Option.join([opt1, opt2, opt3], '/');
```

Makes the code a little easier to read. Couldn't find a linter to enable for it though.

Reply:
Removed the type annotation where possible :)